nilearn/maskers/nifti_labels_masker.py

```diff
@@ -521,11 +521,6 @@ def fit(
             This parameter is unused. It is solely included for scikit-learn
             compatibility.
         """
-        if self.labels_img is None:
-            raise TypeError(
-                "Please provide a valid Nifti-like object for 'labels_img'."
-            )
-
```

Comment:
removes this that I just introduced as _utils.check_niimg_3d used a bit later will do the same check